src/utstring.h

```diff
@@ -121,6 +121,24 @@ do {                                                             \
 
 #define utstring_body(s) ((s)->d)
 
+#define utstring_adopt(s,b,l)                              \
+do {                                                       \
+  if (s) {                                                 \
+    (s)->d = (b);                                          \
+    (s)->n = (l);                                          \
+    (s)->i = (l);                                          \
+  }                                                        \
+} while(0)
+
+#define utstring_disown(s,b)                               \
+do {                                                       \
+  if (s) {                                                 \
+    (b) = (s)->d;                                          \
+    utstring_clear(s);                                     \
```

Comment:
This is incorrect; `utstring_clear` will set `s->d[0] = '\0'`, blowing away the contents of `b`. What you want here is more like `utstring_init(s)`, but actually calling out to `utstring_init` seems confusing to me; I think you should just inline the couple of statements you want here. Which IMO would be

    (b) = (s)->d;
    (s)->d = NULL;
    (s)->i = 0;
    (s)->n = 0;